temporal-sdk/src/main/java/io/temporal/internal/sync/WorkflowInternal.java

```diff
@@ -520,6 +506,27 @@ public static void sleep(Duration duration) {
     getWorkflowOutboundInterceptor().sleep(duration);
   }
 
+  public static boolean isWorkflowThread() {
+    return WorkflowThreadMarker.isWorkflowThread();
+  }
+
+  public static <T> T deadlockDetectorOff(Functions.Func<T> func) {
+    if (isWorkflowThread()) {
+      Runnable deadlockDetectorUnlockHandle = getWorkflowThread().lockDeadlockDetector();
```

Comment:
Arguably something that returns `Closeable` here so you could use try-with-resources would be clearer

Reply:
I can do AutoCloseable. I don't want to make this thingy idempotent, because it will increase synchronization costs without real benefit.